src/server.cpp

```diff
@@ -1671,17 +1671,56 @@ void Server::SendHUDSetParam(session_t peer_id, u16 param, const std::string &va
 	Send(&pkt);
 }
 
-void Server::SendSetSky(session_t peer_id, const video::SColor &bgcolor,
-		const std::string &type, const std::vector<std::string> &params,
-		bool &clouds)
+void Server::SendSetSky(session_t peer_id, const SkyboxParams &params)
 {
 	NetworkPacket pkt(TOCLIENT_SET_SKY, 0, peer_id);
-	pkt << bgcolor << type << (u16) params.size();
 
-	for (const std::string &param : params)
-		pkt << param;
+	pkt << params.bgcolor << params.type
+		<< params.clouds;
```

Comment:
there's zero backwards-compatibility here

Reply:
What if the server sent a TOCLIENT_SET_SUN or TOCLIENT_SET_STARS packet? That would cause a client error trying to figure out said packet. API compatible was a better choice than packet compatible.

Reply:
Now backwards compatible upto 5.0.[0/1] clients.